alerting/src/main/kotlin/org/opensearch/alerting/transport/SecureTransportAction.kt

```diff
@@ -76,18 +76,12 @@ interface SecureTransportAction {
     }
 
     fun <T : Any> validateUserBackendRoles(user: User?, actionListener: ActionListener<T>): Boolean {
+        if (!filterByEnabled || user == null) {
+            return true
+        }
+
         if (filterByEnabled) {
```

Comment:
I think we maybe have an issue on alerting when starting the transport action when security is enabled and filterBy also.
Just wanted to check with you what's your opinion.

Namely, when starting alerting transport actions (like for example TransportIndexMonitorAction: https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt#L126-L128)
we are calling this check: if (!validateUserBackendRoles(user, actionListener)) {
return
}

This check first checks if the the filterBy is enabled and then check if the user is null -- if both of them are true - the error is raised.
Check it out here: https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/SecureTransportAction.kt#L79-L82

On other plugins, for example on notification, this check is slightly different:
https://github.com/opensearch-project/notifications/blob/main/notifications/notifications/src/main/kotlin/org/opensearch/notifications/security/UserAccessManager.kt#L55-L57
or for example on index-management:
https://github.com/opensearch-project/index-management/blob/main/src/main/kotlin/org/opensearch/indexmanagement/util/SecurityUtils.kt#L137-L139

ie. it checks if the user is null OR if the filterBy is enabled and returns true.
The check we have on alerting will prevent us, to call one transport action from another with stashed context - since it will raise an error (because when the context is stashed the user will be null)

Should we adjust the check on alerting to reflect the same logic or? Tnx in advance!
@eirsep @lezzago